Adyen/Analytics/AnalyticsProvider/AnalyticsProvider.swift

```diff
@@ -45,6 +46,10 @@ internal final class AnalyticsProvider: AnalyticsProviderProtocol {
     
     private var batchTimer: Timer?
     private let batchInterval: TimeInterval
+    
+    private var isValidAttemptId: Bool {
```

Comment:
Doesn't it start turning into a whole logic around this id?
Maybe an enum could be a simple logic wrapper:
```
enum CheckoutAttempt {
    case id(String)
    case doNotTrack

    var isValid: Bool
}
```